src/components/utils/test/thread_test.cc

```diff
@@ -296,6 +302,101 @@ TEST_F(ThreadOptionsTest, StartOneThreadTwice_ExpectTheSameThreadStartedTwice) {
   EXPECT_TRUE(delegate_->check_value());
 }
 
+
+TEST(ThreadExecutionFlowTest, Start_Success) {
+  // Create a delegate with infinite threadMain
+  InfiniteMainTestDelegate* delegate = new InfiniteMainTestDelegate();
+  Thread* thread = CreateThread(kThreadName.c_str(), delegate);
+  EXPECT_TRUE(thread->start(threads::ThreadOptions()));
+  EXPECT_TRUE(thread->is_joinable());
+
+  // Wait for the thread to notify it has started execution
+  WaitForThreadStart(delegate);
```

Comment:
@wolfylambova22 I think it would be good to use here and in similar test cases somwthing like `WaitFor` with timeout. To avoid entire system lock until thread actual start because its start depends on OS.